spec/requests/providers/applicant_details_spec.rb

```diff
@@ -0,0 +1,150 @@
+require 'rails_helper'
+
+RSpec.describe Providers::ApplicantDetailsController, type: :request do
+  let(:application) { create(:legal_aid_application) }
+  let(:application_id) { application.id }
+  let(:provider) { application.provider }
+
+  describe 'GET /providers/applications/:legal_aid_application_id/applicant_details' do
+    subject { get "/providers/applications/#{application_id}/applicant_details" }
+
+    context 'when the provider is not authenticated' do
+      before { subject }
+      it_behaves_like 'a provider not authenticated'
+    end
+
+    context 'when the provider is authenticated' do
+      before do
+        login_as provider
+      end
+
+      it 'returns http success' do
+        subject
+        expect(response).to have_http_status(:ok)
+      end
+
+      context 'has already got applicant info' do
+        let(:applicant) { create(:applicant) }
+        let(:application) { create(:legal_aid_application, applicant: applicant) }
+
+        it 'display first_name' do
+          subject
+          expect(unescaped_response_body).to include(applicant.first_name)
+        end
+      end
+    end
+  end
+
+  describe 'PATCH /providers/applications/:legal_aid_application_id/applicant_details' do
+    let(:params) do
+      {
+        applicant: {
+          first_name: 'John',
+          last_name: 'Doe',
+          national_insurance_number: 'AA 12 34 56 C',
+          dob_year: '1981',
+          dob_month: '07',
+          dob_day: '11',
+          email: Faker::Internet.safe_email
```

Comment:
Could `attributes_for(:applicant)` be used here?

Reply:
Indeed. I'll change it. 
(This code is not new but copied from the previous `applicants_spec.rb` file)